src/mito2/src/region/options.rs

```diff
@@ -54,6 +67,9 @@ pub struct RegionOptions {
     pub index_options: IndexOptions,
     /// Memtable options.
     pub memtable: Option<MemtableOptions>,
+    /// The mode to merge duplicate rows.
+    /// Only takes effect when `append_mode` is `false`.
+    pub merge_mode: MergeMode,
```

Comment:
I think it's better to validate this constraint when creating or opening regions.